test/erc_to_native/foreign_bridge.test.js

```diff
@@ -358,6 +378,127 @@ contract('ForeignBridge_ERC20_to_Native', async accounts => {
+      const signature = await sign(authorities[0], message)
+      const vrs = signatureToVRS(signature)
+      const oneSignature = packSignatures([vrs])
+      false.should.be.equal(await foreignBridge.relayedMessages(transactionHash))
+
+      // wait for a small interest in DSR
+      await delay(1500)
+
+      const { logs } = await foreignBridge.executeSignatures(message, oneSignature).should.be.fulfilled
+
+      logs[0].event.should.be.equal('RelayedMessage')
+      logs[0].args.recipient.should.be.equal(recipientAccount)
+      logs[0].args.value.should.be.bignumber.equal(value)
+      const balanceAfter = await token.balanceOf(recipientAccount)
+      const balanceAfterBridge = await token.balanceOf(foreignBridge.address)
+      balanceAfter.should.be.bignumber.equal(balanceBefore.add(value))
+      balanceAfterBridge.should.be.bignumber.equal(ether('0.1'))
+      true.should.be.equal(await foreignBridge.relayedMessages(transactionHash))
+    })
+
+    it('should executeSignatures with enabled chai token, not enough dai, high dai limit', async () => {
+      await foreignBridge.initializeChaiToken({ from: owner })
+      await token.mint(foreignBridge.address, ether('1'), { from: owner })
+      await foreignBridge.setMinDaiTokenBalance(ether('0.1'), { from: owner })
+      await foreignBridge.convertDaiToChai()
+      // in case of high limit, bridge should withdraw all invested tokens
+      await foreignBridge.setMinDaiTokenBalance(ether('5'), { from: owner })
+      expect(await chaiToken.balanceOf(foreignBridge.address)).to.be.bignumber.gt(ZERO)
+      expect(await token.balanceOf(foreignBridge.address)).to.be.bignumber.gte(ether('0.1'))
+      expect(await token.balanceOf(foreignBridge.address)).to.be.bignumber.lte(ether('0.11'))
+
+      const recipientAccount = accounts[3]
+      const balanceBefore = await token.balanceOf(recipientAccount)
+
+      const transactionHash = '0x1045bfe274b88120a6b1e5d01b5ec00ab5d01098346e90e7c7a3c9b8f0181c80'
+      const message = createMessage(recipientAccount, value, transactionHash, foreignBridge.address)
+      const signature = await sign(authorities[0], message)
+      const vrs = signatureToVRS(signature)
+      const oneSignature = packSignatures([vrs])
+      false.should.be.equal(await foreignBridge.relayedMessages(transactionHash))
+
+      // wait for a small interest in DSR
+      await delay(1500)
+
+      const { logs } = await foreignBridge.executeSignatures(message, oneSignature).should.be.fulfilled
+
+      logs[0].event.should.be.equal('RelayedMessage')
+      logs[0].args.recipient.should.be.equal(recipientAccount)
+      logs[0].args.value.should.be.bignumber.equal(value)
+      const balanceAfter = await token.balanceOf(recipientAccount)
+      const balanceAfterBridge = await token.balanceOf(foreignBridge.address)
+      balanceAfter.should.be.bignumber.equal(balanceBefore.add(value))
+      balanceAfterBridge.should.be.bignumber.gte(ether('1'))
```

Comment:
Is it correct that in this scenario all the tokens invested were converted from chai to dai when executing the signatures? Does it make sense to also check that the chai balance of the bridge should be `ZERO`?

Reply:
Yeah, it makes sense

Reply:
Actually, the chai balance won't be 0, a small interest will be already collected between calls to `convertDaiToChai()` and `executeSignatures()`.
Used a comparison with a relatively small number for that.